### PR TITLE
Created a GameJamManagerContract

### PR DESCRIPTION
**Pull request by Ryan-Gordon**
A GameJamManager is a contract which tracks 1 or more GameJams.

PR Features:
+ Allows a manager to create a new GameJam
+ RBAC for main operations
+ Interfaces used for Contract Registry
+ Updated migrations to use new contracts

Currently has just 1 meaningful test.

### CODE REVIEW

File: contracts/GameJamManager.sol [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+contract GameJamManager {
+    using Roles for Roles.Role;
+
+    // Define a ContractRegistry to be setup in the Constructor
+    IRegistry registry;
+    // Define a mapping to track GameJams using their name with their address
+    mapping (bytes32 => address) public gameJamList;
+
+    // Define RBAC List, Modifiers and Checks
+    Roles.Role private _gameJamManager;
```
Comment (Ryan-Gordon) [inline contracts/GameJamManager.sol]: Should we be looking to extract the common Role logic that is in each contract or is it fine this way ?

File: contracts/GameJamManager.sol [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        _addGameJamManager(msg.sender);
+    }
+
+    //Function to create a new GameJam contract
+    function createGameJam(uint prizeBalance)
+    private
+    onlyGameJamManager
+    returns (address _gameJamAddress)
+    {
+        return address((new GameJam).value(0)(prizeBalance));
```
Comment (Ryan-Gordon) [inline contracts/GameJamManager.sol]: `(new GameJam).value(0)(prizeBalance)` is what we need to invoke a `payable` constructor
Keep that one handy in the notes

File: contracts/GameJamManager.sol (hunk 1 above)
Comment (PaddyMc) [reply contracts/GameJamManager.sol]: It's fine this way, I think we're using the abstraction that `openzeppelin-solidity` built

File: contracts/GameJamManager.sol (hunk 2 above)
Comment (PaddyMc) [reply contracts/GameJamManager.sol]: Yeah please! <3

Comment (PaddyMc) [reply contracts/GameJamManager.sol]: Tbh, I'd like to merge this in as is and I'll open an PR later today with some stuff ❤️

**Pull request merged**